processing/src/main/java/io/druid/query/timeseries/TimeseriesQueryQueryToolChest.java

```diff
@@ -125,6 +130,71 @@ public TimeseriesQueryQueryToolChest(
         );
       }
     };
+
+    return new QueryRunner<Result<TimeseriesResultValue>>()
+    {
+      @Override
+      public Sequence<Result<TimeseriesResultValue>> run(
+          final QueryPlus<Result<TimeseriesResultValue>> queryPlus,
+          final Map<String, Object> responseContext
+      )
+      {
+        final TimeseriesQuery query = (TimeseriesQuery) queryPlus.getQuery();
+        final Sequence<Result<TimeseriesResultValue>> baseResults = resultMergeQueryRunner.run(
+            queryPlus.withQuery(
+                queryPlus.getQuery()
+                         .withOverriddenContext(
+                             ImmutableMap.of(TimeseriesQuery.CTX_GRAND_TOTAL, false)
+                         )
+            ),
+            responseContext
+        );
+
+        if (query.isGrandTotal()) {
```

Comment:
@gianm wouldn't this materialize the sequence before actual streaming of the results starts?

Reply:
I don't think so, because `Sequences.map` is lazy.